src/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java

```diff
@@ -428,18 +470,43 @@ private void recoverBlockedListeners() {
         }
     }
 
-    private void recoverConnection() throws IOException, InterruptedException {
-        boolean recovering = true;
-        while (recovering) {
+	//Returns true if the connection was recovered, 
+	//false if application initiated shutdown while attempting recovery.  
+    private boolean recoverConnection() throws IOException, InterruptedException {
+        while (!manuallyClosed) //manuallyClosed marked volatile -ensures thread cannot cache value of manuallyClosed.
+		{
             try {
-                this.delegate = this.cf.newConnection();
-                recovering = false;
+				RecoveryAwareAMQConnection newConn = this.cf.newConnection();
+				synchronized(recoveryLock)
+				{
+					if (!manuallyClosed)
+					{
+						//This is the standard case.
+						this.delegate = newConn;					
+						return true;
+					}
+				}
+
+				//This is the once in a blue moon case.  
+				//Application code just called close as the connection
+				//was being re-established.  So we attempt to close the newly created connection.
+				try
+				{
+					newConn.close(); //Is there a standard error code/shutdown message to use here? 
```

Comment:
If you want to close the delegate and ignore all exceptions, use `Connection#abort()`.